packages/legacy/core/App/screens/PasteUrl.tsx

```diff
@@ -0,0 +1,139 @@
+import { testIdWithKey } from '../utils/testable'
+
+export type PasteProps = StackScreenProps<ConnectStackParams>
+
+const PasteUrl: React.FC<PasteProps> = ({ navigation }) => {
+  const { ColorPallet, TextTheme } = useTheme()
+  const [pastedContent, setPastedContent] = useState<string>('')
+  const [errorMessage, setErrorMessage] = useState<{ title: string; message: string } | undefined>()
+  const { t } = useTranslation()
+  const { agent } = useAgent()
+  const container = useContainer()
+  const logger = container.resolve(TOKENS.UTIL_LOGGER)
+  const { enableImplicitInvitations, enableReuseConnections } = useConfiguration()
+  const styles = StyleSheet.create({
+    container: {
+      flex: 1,
+      backgroundColor: ColorPallet.brand.primaryBackground,
+    },
+    content: {
+      margin: 20,
+    },
+    description: {
+      ...TextTheme.normal,
+      marginBottom: 20,
+    },
+    textBox: {
+      ...TextTheme.normal,
+      textAlignVertical: 'top',
+      borderColor: ColorPallet.grayscale.darkGrey,
+      borderWidth: 1,
+      borderRadius: 5,
+      backgroundColor: ColorPallet.grayscale.lightGrey,
+    },
+    buttonContainer: {
+      margin: 20,
+    },
+  })
+
+  const processPastedContent = async () => {
+    try {
+      await connectFromScanOrDeepLink(
+        pastedContent,
+        agent,
+        logger,
+        navigation?.getParent(),
+        false, // isDeepLink
+        enableImplicitInvitations,
+        enableReuseConnections
+      )
+    } catch (err: unknown) {
+      setErrorMessage({ title: t('PasteUrl.ErrorInvalidUrl'), message: t('PasteUrl.ErrorInvalidUrlDescription') })
+    }
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <ScrollView>
+        <Modal visible={!!errorMessage} testID={testIdWithKey('ErrorModal')} animationType="fade" transparent>
+          <View
+            style={{
+              flex: 1,
+              paddingHorizontal: 10,
+              justifyContent: 'center',
+              alignItems: 'center',
+              backgroundColor: 'rgba(0,0,0,0.6)',
+            }}
+          >
+            <InfoBox
+              notificationType={InfoBoxType.Error}
+              title={errorMessage?.title ?? ''}
+              description={errorMessage?.message ?? ''}
+              onCallToActionPressed={() => setErrorMessage(undefined)}
+              onCallToActionLabel={t('Global.TryAgain')}
+            />
+          </View>
+        </Modal>
+        <View style={styles.content}>
+          <Text style={styles.description}>{t('PasteUrl.PasteUrlDescription')}</Text>
+          <TextInput
+            testID={testIdWithKey('PastedUrl')}
+            style={styles.textBox}
+            numberOfLines={15}
+            multiline
+            value={pastedContent}
+            onChangeText={(text) => setPastedContent(text)}
+          />
+        </View>
+        <View style={styles.buttonContainer}>
+          <TouchableOpacity
```

Comment:
Why add a `Button` inside a `TouchableOpacity`?

Reply:
the wireframes want a popup to appear when the user clicks on the disabled continue button with an empty text box. The button is disabled when the textbox is empty so I had to add a TouchableOpacity in order to detect a button press when the button is disabled

Reply:
it's a bit hacky I admit, but I haven't had to make a disabled button respond to anything before, there might be a way to do it with a <View> instead but that's essentially the same as a touchable opacity just without a bit of styling